engine/src/build/config/darwin/BUILD.gn

```diff
@@ -32,21 +32,21 @@ config("compiler") {
     # x86, arm, armv7, etc.
     assert(false, "unsupported cpu: $current_cpu")
   }
-  # Environment.
-  if (use_ios_simulator) {
-    _triplet_environment = "-simulator"
-  } else {
-    _triplet_environment = ""
-  }
-  # OS.
+  # OS and deployment target.
   if (is_ios) {
     _triplet_os = "apple-ios"
+    _deployment_target = ios_deployment_target
   } else if (is_mac) {
     _triplet_os = "apple-macos"
+    _deployment_target = mac_deployment_target
+  }
+  # Environment.
+  if (use_ios_simulator) {
+    _triplet_environment = "-simulator"
   } else {
-    assert(false, "unsupported Apple os")
+    _triplet_environment = ""
   }
-  _target_triplet = "$_triplet_cpu-$_triplet_os$mac_deployment_target$_triplet_environment"
+  _target_triplet = "$_triplet_cpu-$_triplet_os$_deployment_target$_triplet_environment"
```

Comment:
sanity check: there's a `-` between cpu & os, but not other places?

Reply:
See line 45 above; there's a `-` between `apple-ios` and `simulator` if it's a simulator build. So the full "triplet" is `arm64-apple-ios-simulator` in that case. Otherwise it's just `arm64-apple-ios`.

Triplet is a loosely defined historical term that compilers use for this. https://wiki.osdev.org/Target_Triplet